src/one-source/buffer-with-time-or-count.js

```diff
@@ -0,0 +1,63 @@
+const {createStream, createProperty} = require('../patterns/one-source');
+
+const mixin = {
+
+  _init({wait, count, flushOnEnd}) {
+    this._wait = wait;
+    this._count = count;
+    this._flushOnEnd = flushOnEnd;
+    this._intervalId = null;
+    this._$onTick = () => this._flush();
+    this._buff = [];
+  },
+
+  _free() {
+    this._$onTick = null;
+    this._buff = null;
+  },
+
+  _flush() {
+    if (this._buff !== null && this._buff.length !== 0) {
```

Comment:
Hm, I think you're right. That makes sense to do check that `.lenght !== 0` in "on end" flush. I can't imagine a use case where it would be necessary to always have that "on end" flush. And we actually do that check in `.bufferWhile()` (we always check in there, but it can be empty only on end).